scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -756,6 +759,11 @@ def apply_filters(self, **kwargs):
                     if ts.platform_exclude and plat.name in ts.platform_exclude:
                         instance.add_filter("In test case platform exclude", Filters.TESTSUITE)
 
+                    if not device_testing \
+                            and ts.simulation_exclude \
+                            and plat.simulation in ts.simulation_exclude:
+                        instance.add_filter("In test case simulation exclude", Filters.SIMULATION)
```

Comment:
My point is, that maybe you can check if the satisfying result would be obtained without introduction of new yaml entry (simulation_exclude) but with using the already existing one from test yamls: platform_type: mcu. Something like

```suggestion
                    if not device_testing and ts.platform_type == ['mcu']:
                        instance.add_filter("Test is made solely for on-target execution", Filters.SIMULATION)
```

Reply:
Wouldn't this filter out `ts.platform_type == ['mcu']` as long as not testing on physical boards?

Reply:
I believe this is the point of `platform_type` with only `mcu` inside, stating that the test is meant only for mcu. My concern is that with filtering I proposed such tests won't also build without "device-testing". But on the other hand just building, for such tests doesn't bring much value IMO. BTW it seems there are only 9 cases, where "platform_type" is used in test yaml, and this particular (kernel.timer) is the only one which has only 'mcu' listed.

Reply:
> I believe this is the point of `platform_type` with only `mcu` inside

I think I get what you mean now..

Reply:
I [tried this direction today](https://github.com/zephyrproject-rtos/zephyr/compare/main...ycsin:pr/twister_simulation_exclude1?expand=1) but it's confusing to assume that `type: mcu` means 'real hardware', given that it might have `qemu` or `renode` as its underlying simulator. Forcing tests to run only on real hardware when `platform_type` == `['mcu']` would affect other simulator(s) that could have passed the test.